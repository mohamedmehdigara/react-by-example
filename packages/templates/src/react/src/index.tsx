import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
