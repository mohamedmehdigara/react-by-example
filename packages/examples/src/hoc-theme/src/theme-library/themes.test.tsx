import { cleanup, render, RenderResult } from '@testing-library/react'
import React, { FC, useContext } from 'react'
import { ThemeContext } from './internal/theme-context'
import { DarkTheme, LightTheme } from './themes'

describe('themes', () => {
  let component: RenderResult

  const Test: FC = () => {
    const theme = useContext(ThemeContext)

    return <pre>{JSON.stringify(theme, null, '  ')}</pre>
  }

  afterEach(() => {
    cleanup()
  })

  describe('dark', () => {
    beforeEach(() => {
      component = render(
        <DarkTheme>
          <Test />
        </DarkTheme>,
      )
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('light', () => {
    beforeEach(() => {
      component = render(
        <LightTheme>
          <Test />
        </LightTheme>,
      )
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })
})
