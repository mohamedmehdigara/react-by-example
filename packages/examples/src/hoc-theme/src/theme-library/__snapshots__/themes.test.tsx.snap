// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes dark should snapshot 1`] = `
<div>
  <pre>
    {
  "color": {
    "background": "#282a36",
    "foreground": "#f8f8f2"
  }
}
  </pre>
</div>
`;

exports[`themes light should snapshot 1`] = `
<div>
  <pre>
    {
  "color": {
    "background": "#f8f8f2",
    "foreground": "#282a36"
  }
}
  </pre>
</div>
`;
